backend/LexBoxApi/Services/CrdtCommitService.cs

```diff
@@ -13,31 +13,49 @@ public async Task AddCommits(Guid projectId, IAsyncEnumerable<ServerCommit> comm
     {
         await using var transaction = await dbContext.Database.BeginTransactionAsync();
         var commitsTable = dbContext.CreateLinqToDBContext().GetTable<ServerCommit>();
-        await foreach (var commitChunk in commits.Chunk(100))
+        const int commitsThreshold = 100;
+        const int changeThreshold = 150;
+        var currentChangeCount = 0;
+        var commitBucket = new List<ServerCommit>(100);
+        await foreach (var serverCommit in commits)
+        {
+            commitBucket.Add(serverCommit);
+            currentChangeCount += serverCommit.ChangeEntities.Count;
+            if (currentChangeCount < changeThreshold && commitBucket.Count < commitsThreshold)
+            {
+                continue;
+            }
+
+            await FlushCommits();
+        }
+
+        if (commitBucket.Count > 0) await FlushCommits();
+
+        await transaction.CommitAsync();
+
+        async Task FlushCommits()
         {
-            //using merge instead of BulkCopy to support skipping inserts of commits that already exist
```

Comment:
I found this to be an informative comment

Reply:
This code has been touched a lot in https://github.com/sillsdev/languageforge-lexbox/pull/1608